airflow/www/security.py

```diff
@@ -348,19 +348,21 @@ def can_read_dag(self, dag_id, user=None) -> bool:
         if not user:
             user = g.user
         dag_resource_name = permissions.resource_name_for_dag(dag_id)
-        return self._has_view_access(
-            user, permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG
-        ) or self._has_view_access(user, permissions.ACTION_CAN_READ, dag_resource_name)
+        return bool(
+            self._has_view_access(user, permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG)
+            or self._has_view_access(user, permissions.ACTION_CAN_READ, dag_resource_name)
+        )
```

Comment:
All the changes in this file should be unnecssary -- can you revert them please?

Reply:
this is a problem as `_has_view_access` doesn't always return a bool, it returns a NoneType and then fails later on - https://github.com/apache/airflow/pull/9973#discussion_r614169639

Reply:
And why is this a problem? None evaluates falsey in a boolean context.

Reply:
Ah, see https://github.com/apache/airflow/pull/9973#discussion_r629501511